walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/OverlappingTimeRange.java

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Wrapper class representing a span of time that contains overlapping time
+ * ranges.
+ */
+public final class OverlappingTimeRange {
+  private List<TimeRange> overlappingTimes = new ArrayList<>();
```

Comment:
I recommend using ImmutableList when possible.

"For field types, you should generally use the immutable type (such as ImmutableList) instead of the general collection interface type (such as List). This communicates within the class the semantic guarantees of immutable types, which is almost always very useful information. When it's a general collection type, future editors of the class may be under the impression that they can mutate the field, leading to failures at runtime."